packages/core/src/utils/assert-sign-in-experience.ts

```diff
@@ -0,0 +1,26 @@
+import { Branding, BrandingStyle, TermsOfUse } from '@logto/schemas';
+
+import assertThat from '@/utils/assert-that';
+
+export const assertBranding = (branding?: Branding) => {
+  if (!branding) {
+    return;
+  }
```

Comment:
same.
the whole`assertBranding` can be written as a single`assertThat` assertion.

Reply:
Write in a single assertThat like this:
```ts
export const assertBranding = (branding?: Branding) => {
  assertThat(
    !branding ||
      branding.style !== BrandingStyle.Logo_Slogan ||
      (branding.slogan && branding.slogan.trim() === ''),
    'sign_in_experiences.empty_slogan'
  );
};
```
Since it's less readable, current implementation is better.

Reply:
Updated. See Slack [thread](https://silverhand-io.slack.com/archives/C02A8G4HVAM/p1648713165060019)

```ts
export const assertBranding = (branding?: Branding) => {
  if (!branding) {
    return;
  }

  if (branding.style === BrandingStyle.Logo_Slogan) {
    assertThat(branding.slogan?.trim(), 'sign_in_experiences.empty_slogan');
  }
};
```